src/app/(nextjs_migration)/(authenticated)/user/welcome/scanning/page.tsx

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Image from "next/image";
+import Script from "next/script";
+
+import { getL10n } from "../../../../../functions/server/l10n";
+import { getLocale } from "../../../../../../utils/fluent.js";
+
+export async function generateMetadata() {
+  const l10n = getL10n();
+  return {
+    title: "Welcome - Scanning", // FIXME l10n.getString("breach-meta-title"),
+    twitter: {
+      card: "summary_large_image",
+      title: l10n.getString("brand-fx-monitor"),
+      description: l10n.getString("meta-desc-2"),
+      images: ["/images/og-image.webp"],
+    },
+    openGraph: {
+      title: l10n.getString("brand-fx-monitor"),
+      description: l10n.getString("meta-desc-2"),
+      siteName: l10n.getString("brand-fx-monitor"),
+      type: "website",
+      url: process.env.SERVER_URL,
+      images: ["/images/og-image.webp"],
+    },
+  };
+}
+
+export default async function UserWelcomeScanning() {
+  const l10n = getL10n();
+
+  const current = 1;
+  const total = 672;
+  const percentage = ((current / total) * 100).toFixed(1);
+
+  const body = JSON.stringify({
+    first_name: "Test",
+    last_name: "User",
+    addresses: [
+      {
+        state: "NY",
+        city: "New York",
+        zip: "11111",
+        address_line: "1st Ave 1 Apt 1",
+      },
+    ],
+  });
+
+  async function callOneRep(method: string, path: string) {
+    const bearerToken = process.env.ONEREP_API_KEY;
+    const options = {
+      method,
+      headers: {
+        Authorization: `Bearer ${bearerToken}`,
+        "Content-Type": "application/json",
+      },
+    };
+    if (method !== "GET" && method !== "HEAD") {
+      //@ts-ignore FIXME
+      options.body = body;
+    }
+    const result = await fetch(`https://api.onerep.com/${path}`, options);
+    if (!result.ok) {
+      throw new Error("Error connecting to provider");
+    }
+    return result.json();
+  }
+
+  const profile = await callOneRep("POST", "profiles");
+  const scan = await callOneRep("POST", `profiles/${profile.id}/scans`);
```

Comment:
Q: is it possible that `profile` here can be `null`|`undefined` (causing the `profile.id` to error)?

Reply:
In general I'd prefer to not catch errors (preferring to let a global error handler catch them) unless we're sure we can actually handle them. I don't think there's anything useful we can do if this fails, the user shouldn't be at this page unless they successfully created a profile and a scan and they were both saved in the database. If this sort of thing fails, it likely means the database is unavailable or the network is down, so we can't really provide the user much of an experience at this point.